FWCore/PluginManager/bin/refresh.cc

```diff
@@ -272,9 +272,9 @@ int main(int argc, char** argv) try {
         // Throw if any of the child died with non 0 status.
         int status = 0;
         waitpid(worker, &status, 0);
-        if (WIFEXITED(status) == true && status != 0) {
-          std::cerr << "Error while processing." << std::endl;
-          exit(status);
+        if (WIFEXITED(status) == true) {
```

Comment:
I don't know why the `status != 0` check was there. https://man7.org/linux/man-pages/man3/wait.3p.html has
```
       WIFEXITED(stat_val)
             Evaluates to a non-zero value if status was returned for a
             child process that terminated normally.
```
so I'd think checking `WIFEXITED()` would be sufficient?

Reply:
Shouldn't the check be `if(WIFEXITED(status) == 0) {`?

Reply:
I think the check should be 
```
if (WIFEXITED(status) !=0 and WEXITSTATUS(status) != 0) {
```
as the guarantee is `WIFEXITED` returns a _non zero_ value, not that the value when compared to `true` will evaluate to `true`.

Reply:
The `WIFEXITED(status) != 0` instead of `== true` came to my mind too, and is included in the latest push.